docs/v3tov4.md

```diff
@@ -0,0 +1,340 @@
+  "integration":{                              # this data will be added to all metrics and events as attributes,                                               
+                                               # and also sent as inventory
+    "name":"integration name",
+    "version":"integration version"
+  },
+  "data":[                                    # List of objects containing entities, metrics, events and inventory
+    {
+      "entity":{                              # this object is optional. If it's not provided, then the Entity will get 
+                                              # the same entity ID as the agent that executes the integration. 
+        "name":"redis:192.168.100.200:1234",  # unique entity name per customer account
+        "type":"RedisInstance",               # entity's category
+        "displayName":"my redis instance",    # human readable name
+        "metadata":{}                         # can hold general metadata or tags. Both are key-value pairs that will 
+                                              # be also added as attributes to all metrics and events
+      },
+      "metrics":[                             # list of metrics using the dimensional metric format
+        {
+          "name":"redis.metric1",
+          "type":"count",                     # gauge, count, summary, cumulative-count, rate or cumulative-rate
+          "value":93, 
+          "attributes":{}                     # set of key-value pairs that define the dimensions of the metric
+        }
+      ],
+      "inventory":{...},                      # Inventory remains the same
+      "events":[...]                          # Events remain the same
+    }
+  ]
+}
+```
+
+## GoSDK v4 API changes
+
+This section enumerates the main changes you have to keep in mind to upgrade from GoSDK v3.x to v4.x.
+
+### Entities
+
+In previous versions of the SDK, we had a distinction between the Local Entity and Remote entities. You could get the 
+`Local Entity` calling the method
+
+`func (i *Integration) LocalEntity() *Entity`
+
+or create a `Remote Entity` calling
+
+`func (i *Integration) Entity(name, namespace string, idAttributes ...IDAttribute) (e *Entity, err error)`
+
+In SDK v4, you can obtain the Host Entity or create Entities. The Host Entity is a the entity attached to the host 
+where the agent is running. It inherits the same entity ID as the agent which executes the integration. On the other
+hand, an Entity has its own entity ID. Both types of entities are decorated with the metadata from the host where the
+agent is running.
+
+To get the Host Entity you need to access the `HostEntity` of an integration. Example:
+
+```
+// create the integration
+i, err := integration.New(integrationName, integrationVersion)
+
+// add some inventory to the Host Entity
+i.HostEntity.AddInventoryItem("key", "value", "foo")
+```
+
+To create an Entity you need to call the following method inside  integration
```

Comment:
Maybe "call the following method on the integration struct/object".  "inside integration" sounds a bit weird to me.